app/controllers/posts_controller.rb

```diff
@@ -3,7 +3,38 @@ def index
     @user = User.find(params[:user_id].to_i)
   end
 
+  def new
+    newpost = Post.new
+    respond_to do |format|
+      format.html { render :new, locals: { newpost: } }
+    end
+  end
+
+  def create
+    @post = current_user.posts.new(post_params)
+    @post.likes_counter = 0
+    @post.comments_counter = 0
+
+    respond_to do |format|
+      format.html do
+        if @post.save
+          @post.update_posts_counter
+          flash[:success] = 'Post saved successfully'
+          redirect_to user_post_path(current_user, @post.id)
+        else
+          newpost = Post.new
+          flash.now[:error] = 'Error: post could not be saved'
+          render :new, locals: { newpost: }
+        end
+      end
+    end
+  end
+
   def show
     @post = Post.find(params[:id])
   end
+
+  def post_params
+    params.require(:post).permit(:title, :text)
+  end
 end
```

Comment:
## Nice for using Strong parameters
- I suggest you mention `private` in these private methods. keep applying this in all controllers
- See an Example from this link 👉 [Ruby on Rails Strong parameters](https://guides.rubyonrails.org/action_controller_overview.html#strong-parameters)

Reply:
@OLIPLICHE thanks for mentioning this, I will implement this suggestion